motoman_msgs/srv/SelectTool.srv

```diff
@@ -0,0 +1,92 @@
+#         The motion queue internal to MotoROS caches a number of segments in
+#         a FIFO. Only segments received AFTER this service was invoked will
+#         be executed with the new tool. Any segments received before a tool
+#         switch was invoked will use the old tool.
+#
+#         To guarantee motion will be executed using the new tool, applications
+#         should check the 'in_motion' field (part of the RobotStatus message
+#         published on the 'robot_status' topic by the driver) and invoke the
+#         service when the robot has come to a stop (and the motion queue of
+#         MotoROS is empty). Any subsequent trajectories will use the new tool.
+#
+# NOTE 4: as MotoROS currently only executes joint space trajectories, changing
+#         tool file with this service DOES NOT affect TCP poses while executing
+#         trajectories.
+#
+#         To clarify: the TCP definition part of the tool file is NOT taken
+#         into account when calculating manipulator motion based on incoming
+#         ROS JointTrajectoryAction goals (as JointTrajectory goals do not
+#         include any Cartesian poses, only absolute joint space coordinates
+#         for each axis).
+#
+#         Instead, ROS applications should use different TF frames to define
+#         tool frames on the ROS side and plan their motions with the
+#         appropriate TF frame as the active tool.
+#
+#         This service could then be used to notify the controller of other
+#         tool characteristics, such as weight, CoG and inertia by switching
+#         to a tool file configured with those parameters.
+#
+# NOTE 5: there is currently no way to retrieve the active tool file.
+#
+# Refer also the general Yaskawa Motoman documentation on tool file
+# configuration, selection and use.
+
+
+# Numeric ID of the group the tool file is defined for.
+#
+# NOTE: this MUST correspond to a group ID currently defined and active on the
+#       controller. The ROS service does not check whether the value specified
+#       here is legal. The MotoROS server will however check this, will refuse
+#       to switch to an illegal value and will report this to the ROS driver.
+#       Callers will be notified by 'success' set to 'false' and an appropriate
+#       error message being provided via the 'message' field of the service
+#       response.
+#
+# legal-values: [0, total_nr_of_groups)
+# required: yes (absence-causes-service-failure)
+# default: no-default
+int32 group_number
+
+
+# Numeric ID of the tool file to switch to for the specified group.
+#
+# NOTE: see the NOTE for the 'group_number' field above. Illegal values for
+#       this field will result in failed service invocations similar to what
+#       is described for that field.
+#
+# legal-values: [0, total_nr_of_defined_tool_files)
+# required: yes (absence-causes-service-failure)
+# default: no-default
+int32 tool_number
+
+# TODO: might want to expose 'sequence_number' here as well
```

Comment:
We don't expose `sequence_number` here as it's a SimpleMessage-level piece of information.